src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/DiagnosticCounter.cs

```diff
@@ -68,7 +68,7 @@ public void Dispose()
         /// </summary>
         public void AddMetadata(string key, string value)
         {
-            lock (MyLock)
+            lock (this)
```

Comment:
@stephentoub - I think you guys already had some discussion so I don't want to needlessly back-track, but I thought locking publicly exposed objects in the BCL was something you recommended against? Name was also exposed so the issue wasn't new, but if we were fixing this I had guessed we would do something like this:


```
private object m_lock = new object();
internal object MyLock => m_lock;
```

Reply:
I assume locking on string is particularly problematic because it interns and dedupes so unrelated strings; that have the same value, can be the same object - leading to spooky locking at a distance

Reply:
> locking publicly exposed objects in the BCL was something you recommended against

We do that in number of places. It is not something we try to avoid. Having private locks was required for partial trust so that partial trust code cannot introduce race in a full trust code.